lstchain/high_level/hdu_table.py

```diff
@@ -198,25 +198,35 @@ def create_hdu_index_hdu(file_list, hdu_index_file, overwrite=False):
         hdu_names = [
             "EFFECTIVE AREA",
             "ENERGY DISPERSION",
-            "BACKGROUND",
-            "PSF",
-            "RAD_MAX",
         ]
 
-        for irf in hdu_names:
+        # 0:PRIMARY, 1:EVENTS, 2:GTI, 3:POINTING, 4-:IRF 
+        if hdu_list[4].header["HDUCLAS3"] == "FULL-ENCLOSURE":
+            hdu_names.extend(["BACKGROUND", "PSF"])
+        if "TH_CONT" in hdu_list[4].header:
+            hdu_names.extend("RAD_MAX")
```

Comment:
rad_max is required for point-like IRFs, so I think it would make more sense to check for `"HDUCLAS3" == "POINT-LIKE``.

It would also make more sense to just check all the hdus present in the file one by one instead of a pre-defined list of hdus.

Reply:
I modified the code to check all the hdus present in the file one by one
```
        # 0:PRIMARY, 1:EVENTS, 2:GTI, 3:POINTING, 4-:IRF 
        for hdu in hdu_list[4:]:
            irf_hdu = hdu.header["HDUCLAS4"]
```